### DISCUSSION TITLE
Default connection pool size

### DISCUSSION THREAD

**Discussion by venkateshgorantla** [xetorthio/jedis#2414] [Q&A]
I see Jedis is opening 18 connections in the pool by default. However default connections size is 8 as per Apache commons GenericObjectPoolConfig. Even after setting MaxActive to 100 Jedis is creating 18 connections only. 

Anyone know why 18 connections are being opened by default

**Comment by sazzad16** [comment]
@venkateshgorantla This doesn't seem right.

1. First of all, where are you finding this number 18? In Java or Redis? How?
2. Please check properly that the pool config being used is actually the one you're providing.
3. What are the values of minIdle and maxIdle?
4. Try using clientName and check in CLIENT LIST whether there are _outsiders_.

**Reply by venkateshgorantla** [reply]
Thanks for quick reply

1. First of all, where are you finding this number 18? In Java or Redis? How?
I checked no of connected client in Redis server using redis-cli client list after my java client startup

2. Please check properly that the pool config being used is actually the one you're providing.
i am using spring data redis. the below configuration is set to JedisConnectionFactory
JedisClientConfiguration clientConfiguration = JedisClientConfiguration.builder()
                .readTimeout(CONNECTION_READ_TIMEOUT)
                .connectTimeout(CONNECTION_TIMEOUT)
                .build();
                clientConfiguration.getPoolConfig().get().setMaxTotal(100);
                clientConfiguration.getPoolConfig().get().setMaxIdle(20);

3. What are the values of minIdle and maxIdle?
there is no maxIdle set and minIdle is 20

4. Try using clientName and check in CLIENT LIST whether there are outsiders.
Yes, i did set clientName and all those 18 connections are from the same clientName.

--

**Reply by sazzad16** [reply]
Sorry, I can't help further.

**Comment by yangbodong22011** [comment]
common-pool2 start will create minIdle connection, then create a new connection when there is no available connection in the current connection pool(can't exceed maxtotal), and if the new connection exceeds maxIdle, but it cools down, only maxIdle connections will still be reserved.
